pkg/controller/endpointslice/utils_test.go

```diff
@@ -918,3 +919,92 @@ func newEmptyEndpointSlice(n int, namespace string, endpointMeta endpointMeta, s
 		Endpoints:   []discovery.Endpoint{},
 	}
 }
+
+func TestSupportedServiceAddressType(t *testing.T) {
```

Comment:
add test for headless services?

Reply:
Any service that does not have a ClusterIP or ClusterIPs set (e.g. families only) is exactly like headless. There are plenty of test coverage for those.

Reply:
based on earlier comment re headless. I am adding new tests for headless (without families).